### DISCUSSION TITLE
Custom GraphQL resolver - Cannot read property 'user' of undefined

### DISCUSSION THREAD

**Discussion by prasanna-rkumar** [strapi/strapi#7992] [Legacy Topics]
I created a custom GraphQL query and a custom controller as its resolver.  
**schema.graphql**.   
`module.exports = {
  query: 'myChild: [Student]!',  
  resolver: {  
    Query: {  
        myChild: strapi.controllers.student.myChild  
    },  
  },  
};`  
**controller code**  
`module.exports = {
	async myChild(obj, options, ctx) {
		const students = await strapi.services.student.find({
			parent_user: ctx.state.user.id
		});
		return students;
	}
};`  
**error** - Cannot read property 'user' of undefined.  
**Expected behavior** - How can I get the authenticated user context (or in my case at least the ObjectId of the logged in user) inside the custom resolver?

**Comment by prasanna-rkumar** [comment]
Refer https://github.com/strapi/strapi/issues/5531 and https://github.com/strapi/strapi/pull/5532

**Reply by lauriejim** [reply]
Thank you for these references 👍 
Thanks to take the time to search, really appreciate it.
